src/followers/render.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * Server-side rendering of the followers block.
+ *
+ * @package Activitypub
+ */
+
+use Activitypub\Blocks;
+use Activitypub\Collection\Actors;
+use Activitypub\Collection\Followers;
+
+/* @var array $attributes Block attributes. */
+$attributes = wp_parse_args( $attributes );
+
+/* @var string $content Inner blocks content. */
+$content = $content ?? '';
+if ( empty( $content ) ) {
+	// Fallback for v1.0.0 blocks.
+	$_title  = $attributes['title'] ?? __( 'Fediverse Followers', 'activitypub' );
+	$content = '<h3 class="wp-block-heading">' . esc_html( $_title ) . '</h3>';
+	unset( $attributes['title'], $attributes['className'] );
+} else {
+	// Remove the outer div if it exists, to avoid nesting issues with block variation classes.
+	$pattern = '/<div\s+class="wp-block-activitypub-followers[^"]*"[^>]*>(.*?)<\/div>/s';
+	if ( preg_match( $pattern, $content, $matches ) ) {
+		$content = $matches[1];
+	}
+}
+
+$followee_user_id = Blocks::get_user_id( $attributes['selectedUser'] );
+if ( is_null( $followee_user_id ) ) {
+	return '<!-- Followers block: `inherit` mode does not display on this type of page -->';
+}
+
+$user = Actors::get_by_id( $followee_user_id );
+if ( is_wp_error( $user ) ) {
+	return '<!-- Followers block: `' . $followee_user_id . '` not an active ActivityPub user -->';
+}
+
+$_per_page     = absint( $attributes['per_page'] );
+$follower_data = Followers::get_followers_with_count( $followee_user_id, $_per_page );
+
+// Prepare Followers data for the Interactivity API context.
+$followers = array_map(
+	function ( $follower ) {
+		return array(
+			'handle' => '@' . $follower->get_preferred_username(),
+			'icon'   => $follower->get_icon(),
+			'name'   => $follower->get_name(),
```

Comment:
```suggestion
			'name'   => $follower->get_name() ?? $follower->get_preferred_username(),
```
this might be empty... should we fall back to get_preferred_username?